### ISSUE TITLE
UndefinedConversionError is raised with Firestore write trigger

### ISSUE THREAD

**Issue by milkcocoa** [GoogleCloudPlatform/functions-framework-ruby#97]
When I added a Firestore trigger, and receive write events, I got `UndefinedConversionError` errors.

Firestore data contains Japanese characters, for example `name: あああ`.

The following is the error message.

```
#<Encoding::UndefinedConversionError: "\xE3" to UTF-8 in conversion from ASCII-8BIT to UTF-8 to US-ASCII>
/layers/google.ruby.bundle/gems/.bundle/gems/ruby/2.7.0/gems/functions_framework-0.9.0/lib/functions_framework/legacy_event_converter.rb:43:in `encode'
/layers/google.ruby.bundle/gems/.bundle/gems/ruby/2.7.0/gems/functions_framework-0.9.0/lib/functions_framework/legacy_event_converter.rb:43:in `read_input_json' 
/layers/google.ruby.bundle/gems/.bundle/gems/ruby/2.7.0/gems/functions_framework-0.9.0/lib/functions_framework/legacy_event_converter.rb:32:in `decode_rack_env' 
/layers/google.ruby.bundle/gems/.bundle/gems/ruby/2.7.0/gems/functions_framework-0.9.0/lib/functions_framework/server.rb:440:in `decode_event' 
/layers/google.ruby.bundle/gems/.bundle/gems/ruby/2.7.0/gems/functions_framework-0.9.0/lib/functions_framework/server.rb:421:in `call' 
/layers/google.ruby.bundle/gems/.bundle/gems/ruby/2.7.0/gems/puma-4.3.8/lib/puma/server.rb:718:in `handle_request' 
/layers/google.ruby.bundle/gems/.bundle/gems/ruby/2.7.0/gems/puma-4.3.8/lib/puma/server.rb:472:in `process_client' 
/layers/google.ruby.bundle/gems/.bundle/gems/ruby/2.7.0/gems/puma-4.3.8/lib/puma/server.rb:328:in `block in run' 
/layers/google.ruby.bundle/gems/.bundle/gems/ruby/2.7.0/gems/puma-4.3.8/lib/puma/thread_pool.rb:134:in `block in spawn_thread' 
```

**Comment by dazuma** [comment]
Hmm. This suggests that Firestore is sending events where the Content-Type doesn't have the expected charset (or the charset is getting lost when filtering through Cloud Functions). I'll try to find out what's going on there, but in the meantime, we can probably assume that incoming JSON documents are utf-8 encoded if not otherwise specified.

**Comment by milkcocoa** [comment]
@dazuma Thanks for dealing with this issue, and I've tried the latest `main` branch, but I have another error like the following.

```
Rack app ("POST /_ah/push-handlers/pubsub/projects/hd09fbdf0dd8a0533p-tp/topics/cloud-functions-6gr4v4tck54wgjeekopraby64q" - (2002:ae5:6209::)): #<Encoding::UndefinedConversionError: "\xE3" from ASCII-8BIT to UTF-8>
2021-05-27 23:15:55 +0000 Rack app ("POST /_ah/push-handlers/pubsub/projects/hd09fbdf0dd8a0533p-tp/topics/cloud-functions-6gr4v4tck54wgjeekopraby64q" - (2002:ae5:6209::)): #<Encoding::UndefinedConversionError: "\xE3" from ASCII-8BIT to UTF-8>
```

**Comment by dazuma** [comment]
Thanks for all the help on this. Your patch has been added, and is now included in the 0.10.0 gem release.
